provider/resource_service.go

```diff
@@ -646,13 +578,13 @@ func serviceModelFromBase(ctx context.Context, b *models.Base) (m ServiceState,
 		for _, ruleItem := range ruleItems {
 			ruleTF := &RuleState{}
 			diags = append(diags, marshalBasicTypesByTag("json", ruleItem, ruleTF)...)
-			ruleSet = append(ruleSet, ruleTF)
+			ruleSet = append(ruleSet, *ruleTF)
 		}
 		ruleState.Rule = ruleSet
-		m.EntityRules = append(m.EntityRules, ruleState)
+		m.EntityRules = append(m.EntityRules, *ruleState)
```

Comment:
same as above, seems like it's better to declare ruleState as a value instead of a pointer.